paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -202,6 +206,103 @@ def report_waiting_aborted(service, deploy_group):
+            self.sc.post(**kwargs)
+
+    def notify_after_mark(self, ret):
+        if ret == 0:
+            message = (
+                f"*{self.service}* - Marked *{self.commit}* for deployment on *{self.deploy_group}*. "
+                "Will start deploying soon!"
+            )
+            self.sc.post(channels=self.channels, message=message)
+            if self.old_commit is not None and self.commit != self.old_commit:
+                mes = (
+                    "Roll it back at any time with:\n"
+                    f"```paasta rollback --service {self.service} --deploy-group {self.deploy_group} "
+                    f"--commit {self.old_commit}```"
+                )
+                self.sc.post(channels=self.channels, message=mes)
+        else:
+            message = f"*{self.service}* - mark-for-deployment failed on *{self.deploy_group}* for *{self.commit}*."
+            self.sc.post(channels=self.channels, message=message)
+            build_url = os.environ.get('BUILD_URL')
+            if build_url is not None:
+                message = "Please see the jenkins output: {}/console".format(build_url)
+                self.sc.post(channels=self.channels, message=message)
+            else:
+                message = "(Run by {} on {})".format(getpass.getuser(), socket.getfqdn())
+                self.sc.post(channels=self.channels, message=message)
+
+    def notify_after_good_deploy(self):
+        message = f"*{self.service}* - Finished for deployment of *{self.commit}* on *{self.deploy_group}*."
+        self.sc.post(channels=self.channels, message=message)
+        if self.old_commit is not None and self.commit != self.old_commit:
+            mes = (
+                "If you need to roll back, run:\n"
+                f"```paasta rollback --service {self.service} --deploy-group {self.deploy_group} "
+                f"--commit {self.commit}```"
+            )
+            self.sc.post(channels=self.channels, message=mes)
+
+    def notify_after_auto_rollback(self):
+        message = (
+            "*{self.service}* - Deployment of {self.commit} for {self.deploy_group} failed!"
+            "\nAuto-rolling back to {}"
+        )
+        self.sc.post(channels=self.channels, message=message)
+
+    def notify_after_abort(self):
+        message = (
+            f"*{self.service} - Deployment of {self.commit} to {self.deploy_group} aborted.\n"
+            "PaaSTA will keep trying to deploy this code until it is healthy."
+        )
+        self.sc.post(channels=self.channels, message=message)
+        if self.old_commit is not None and self.commit != self.old_commit:
+            mes = (
+                "If you need to roll back, run:\n"
+                f"```paasta rollback --service {self.service} --deploy-group {self.deploy_group} --commit {self.commit}"
+            )
+            self.sc.post(channels=self.channels, message=mes)
+
+
+def validate_commit(commit, git_url):
```

Comment:
I think this code is also copy-pasted into wait_for_deployment.py - you could make that call this function.

Reply:
Hmm yea, will do